File: backend/src/main/java/com/app/helpchef/Model/Recipe.java
package com.app.helpchef.Model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Recipe")
public class Recipe {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "recipeId")
    private Long recipeId;

    @Column(name = "recipeName")
    private String recipeName;

    @ElementCollection
    @Column(name = "ingredietns")
    private List<Ingredient> ingredients;

    @Column(name = "description")
    private String description;

    @Column(name = "rate")
    private double rate;

    @ManyToOne
    @JoinColumn(name = "commentId")
    private Comment comment;

}

File: frontend/help-chef/src/app/recipe.ts
import { Ingredient } from "./ingredient";

export interface Recipe {
    recipeId: number;
    recipeName: string;
    ingredients: Ingredient[];
    description: string;
    rate: number;
    comment: {
        commentId: number;
        nick: {
            userId: number;
            email: string;
            name: string;
            surname: string;
            nick: string;
        };
        content: string;
        commentDate: string;
    };
}